build.psm1

```diff
@@ -489,9 +489,15 @@ Fix steps:
     }
 
     # handle TypeGen
-    if ($TypeGen -or -not (Test-Path "$PSScriptRoot/src/System.Management.Automation/CoreCLR/CorePsTypeCatalog.cs")) {
+    $incFileName = 'powershell'
+    if (-not [string]::IsNullOrEmpty($Runtime)) {
+        # File name must be different for Windows and Linux to allow build on Windows and WSL.
+        # Append the first 3 characters fo the Runtime only because otherwise dotnet will crash due to the separation characters. The underscore is only for better readability.
+        $incFileName = "$($incFileName)_$($Runtime.Substring(0,3))"
+    }
+    if ($TypeGen -or -not (Test-Path "$PSScriptRoot/TypeCatalogGen/$($incFileName).inc")) {
```

Comment:
We use `$incFileName).inc` so many times so make sense add:
```powershell
$incFileName = ($incFileName).inc
```

Reply:
Ok, I have addressed that.